mapie/conformity_scores/residual_conformity_scores.py

```diff
@@ -132,3 +141,275 @@ def get_estimation_distribution(
+    """
+
+    def __init__(
+        self,
+        residual_estimator: Optional[RegressorMixin] = None,
+        prefit: bool = False,
+        split_size: Optional[Union[int, float]] = None,
+        random_state: Optional[Union[int, np.random.RandomState]] = None
+    ) -> None:
+        super().__init__(sym=True, consistency_check=False)
+        self.prefit = prefit
+        self.residual_estimator = residual_estimator
+        self.split_size = split_size
+        self.random_state = random_state
+
+    def _check_estimator(
+        self, estimator: Optional[RegressorMixin] = None
+    ) -> RegressorMixin:
+        """
+        Check if estimator is ``None``,
+        and returns a ``LinearRegression`` instance if necessary.
+        If the ``prefit`` attribute is ``True``,
+        check if estimator is indeed already fitted.
+
+        Parameters
+        ----------
+        estimator: Optional[RegressorMixin]
+            Estimator to check, by default ``None``.
+
+        Returns
+        -------
+        RegressorMixin
+            The estimator itself or a default ``LinearRegression`` instance.
+
+        Raises
+        ------
+        ValueError
+            If the estimator is not ``None``
+            and has no ``fit`` nor ``predict`` methods.
+
+        NotFittedError
+            If the estimator is not fitted
+            and ``prefit`` attribute is ``True``.
+        """
+        if estimator is None:
+            return LinearRegression()
+        else:
+            if not (hasattr(estimator, "fit") and
+                    hasattr(estimator, "predict")):
+                raise ValueError(
+                    "Invalid estimator. "
+                    "Please provide a regressor with fit and predict methods."
+                )
+            if self.prefit:
+                if isinstance(estimator, Pipeline):
+                    check_is_fitted(estimator[-1])
+                else:
+                    check_is_fitted(estimator)
+            return estimator
+
+    def _check_parameters(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        y_pred: ArrayLike
+    ) -> Tuple[NDArray, NDArray, NDArray, RegressorMixin]:
+        """
+        Checks all the parameters of the class. Raises an error if the
+        parameter are not well defined.
+
+        Parameters
+        ----------
+        X: ArrayLike
+            Observed values.
+
+        y: ArrayLike
+            Target values.
+
+        y_pred: ArrayLike
+            Predicted targets.
+
+        Returns
+        -------
+        Tuple[NDArray, NDArray, NDArray, RegressorMixin]
+        Well initiated and typed :
+            - X
+            - y
+            - y_pred
+            - residual_estimator
+        """
+        residual_estimator = self._check_estimator(
+            self.residual_estimator
+        )
+        check_random_state(self.random_state)
+        if self.split_size is None:
+            self.split_size = 0.5
```

Comment:
UPDATE: define instead a `_check_split_size` method that checks the value of this variable. I also suggest not modifying the variable object (`self.*`) directly.

Reply:
`split_size` is used in the function `train_test_split` from sklearn and the check is already done in it

Reply:
I am talking about your check :) you test `if self.split_size is None`, is it done in `train_test_split`?

Reply:
Done